### PR TITLE
Enhancements of assertj module 

### PR DESCRIPTION
**Pull request by krystiankaluzny**
Following features were add to assertj module:
- check validation API, equivalent of `ValidationMatcher`
- check value of XPath expresion API, equivalent of `EvaluateXPathMatcher`

### CODE REVIEW

**Comment by krystiankaluzny** [comment]
I also plan to add compare API (equivalent of CompareMatcher) and after that write something in the user guide.

BTW. Now, maven have problem with TLS in java6 build mode. Downloading artifacts failed with message: `Received fatal alert: protocol_version`.

**Comment by bodewig** [comment]
I've seen the Java6 failures in other builds of mine before, the TLS stack in Java6 is pretty old and doesn't cope with longer keys or "more modern" ciphers (I think it doesn't even support Diffie-Hellman properly). Most likely Maven central has enabled a TLS feature that Java6 lacks. I'll look into it separately.

Many, many thanks!

File: xmlunit-assertj/src/main/java/org/xmlunit/assertj/ValidationAssert.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        this.schema = schema;
+    }
+
+    static ValidationAssert create(Object xmlSource, Object... schemaSources) {
+
+        Assertions.assertThat(xmlSource).isNotNull();
+
+        Assertions.assertThat(schemaSources)
+                .isNotNull()
+                .isNotEmpty()
```
Comment (bodewig) [inline xmlunit-assertj/src/main/java/org/xmlunit/assertj/ValidationAssert.java]: The Hamcrest Matcher allows for an empty list of sources which in the end leads to `JAXPValidator` using https://docs.oracle.com/javase/7/docs/api/javax/xml/validation/SchemaFactory.html#newSchema() - i.e. schema sources are picked up from the document's `xsi:schemaLocation` and `xsi:noSchemaLocation`.

File: xmlunit-assertj/src/main/java/org/xmlunit/assertj/ValueAssert.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+package org.xmlunit.assertj;
+
+import org.assertj.core.api.*;
```
Comment (bodewig) [inline xmlunit-assertj/src/main/java/org/xmlunit/assertj/ValueAssert.java]: just a nit, but if you happen to touch this class again, please expand the `*` import. This is nothing that would prevent me from merging the PR, though :-)

File: xmlunit-assertj/src/main/java/org/xmlunit/assertj/ValidationAssert.java (hunk 1 above)
Comment (krystiankaluzny) [reply xmlunit-assertj/src/main/java/org/xmlunit/assertj/ValidationAssert.java]: I assumed that clients whould know if schemaSources is empty or not and use `assertThat(xml).isValid()` or `assertThat(xml).isValidAgainst(schemaSources)` respectively.
But you have right I shouldn't expect that clients would check length of schemaSources

**Comment by bodewig** [comment]
Thanks a lot!

**Pull request merged**